apps/service_gather/lib/gather/extraction.ex

```diff
@@ -28,28 +28,56 @@ defmodule Gather.Extraction do
   @dialyzer {:nowarn_function, handle_continue: 2}
   @impl GenServer
   def handle_continue(:extract, %{extract: extract} = state) do
-    Logger.debug(fn -> "#{__MODULE__}: Started extraction: #{inspect(extract)}" end)
-
     case extract(extract) do
-      {:ok, _destination_pid} ->
-        Logger.debug(fn -> "#{__MODULE__}: Extraction Completed: #{inspect(extract)}" end)
-        Brook.Event.send(Gather.Application.instance(), extract_end(), "gather", extract)
-        {:stop, :normal, state}
+      {:ok, pid} ->
+        Logger.debug(fn -> "#{__MODULE__}: Started extraction: #{inspect(extract)}" end)
```

Comment:
So this was when it was complete, but now is when the extraction starts?

Reply:
Before the entire meat of an extract happened in that one `extract/1` function. When that function returned, the extract was done (for good or bad). Now `extract/1` just _starts_ the extract process and when that process finishes, it’s up to the shutdown callback to notify the extractor through the new handle_info callbacks.